crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -0,0 +1,696 @@
+            if let Some(additional_data) = &custom_onchain_data.additional_data {
+                custom_data_hashmap.insert(*event_index, additional_data.clone());
+            }
+            quote_id_hashmap.insert(*event_index, custom_onchain_data.quote_id);
+        }
+        let mut events_and_quotes = Vec::new();
+        for event in events.iter() {
+            let EthContractEvent { meta, .. } = event;
+            if let Some(meta) = meta {
+                let event_index = meta_to_event_index(meta);
+                if let Some(quote_id) = quote_id_hashmap.get(&event_index) {
+                    events_and_quotes.push((event.clone(), *quote_id));
+                }
+            }
+        }
+        let onchain_order_data = parse_general_onchain_order_placement_data(
+            &*self.quoter,
+            events_and_quotes,
+            self.domain_separator,
+            self.settlement_contract,
+        )
+        .await?;
+        let data_trouple =
+            onchain_order_data
+                .into_iter()
+                .map(|(event_index, onchain_order_placement, order)| {
+                    (
+                        self.custom_onchain_data_parser
+                            .customized_event_data_for_event_index(
+                                &event_index,
+                                &order,
+                                &custom_data_hashmap,
+                                &onchain_order_placement,
+                            ),
+                        (event_index, onchain_order_placement),
+                        order,
+                    )
+                });
+        Ok(data_trouple.unzip_n_vec())
+    }
+}
+
+async fn parse_general_onchain_order_placement_data(
+    quoter: &dyn OrderQuoting,
+    contract_events_and_quotes_zipped: Vec<(EthContractEvent<ContractEvent>, i64)>,
+    domain_separator: DomainSeparator,
+    settlement_contract: H160,
+) -> Result<Vec<(EventIndex, OnchainOrderPlacement, Order)>> {
+    let futures = contract_events_and_quotes_zipped.into_iter().map(
+        |(EthContractEvent { data, meta }, quote_id)| async move {
+            let meta = match meta {
+                Some(meta) => meta,
+                None => return Err(anyhow!("event without metadata")),
+            };
+            let ContractEvent::OrderPlacement(event) = data;
+            let order_data = convert_onchain_order_placement(
+                quoter,
+                &event,
+                quote_id,
+                domain_separator,
+                settlement_contract,
+            )
+            .await;
+            Ok((meta_to_event_index(&meta), order_data?))
+        },
+    );
+    let onchain_order_placement_data = futures::future::join_all(futures).await;
+    onchain_order_placement_data
+        .into_iter()
+        .filter_map(|data| match data {
+            Err(err) => {
+                tracing::debug!("Error while parsing onchain orders: {:}", err);
```

Comment:
~~Is this expected to happen easily? I expect this to happen if the event throws a malformed order (e.g., neither sell nor buy order). However, this means we are trusting a bad contract and we should at least `warn!`ed about it. Or am I missing some expected failure conditions?~~
I see how users could send bad signatures easily or use a bad quote. I still think it should be at least an `info!`.

Reply:
Info logs are only used to give a quick overview for the observing person. Internal errors, usually we don't info!, as they could spam the high-level overview provided by info! logging